src/App.js

```diff
@@ -3,14 +3,41 @@ import TodoList from './TodoList';
 import AddTodoForm from './AddTodoForm';
 
 
+
 function App() {
+  const [todoList, setTodoList] = React.useState([]);
+  const [isLoading, setIsloading] = React.useState(true);
+
+  React.useEffect(() => new Promise((resolve,reject) => setTimeout( () => 
+  resolve({ data: {todoList: JSON.parse(localStorage.getItem("savedTodoList"))}}), 2000))
+  .then(function(result) {
+    console.log(result);
+    setTodoList(result.data.todoList);//is this correct?
+    setIsloading(false);//is this correct?
+  }),[]);
+
+  React.useEffect(() => {
+    if(!isLoading){//is this correct?
+      localStorage.setItem("savedTodoList", JSON.stringify(todoList));
+  }}, [todoList]);
```

Comment:
Yes line 20 is correct, again discuss with your tutor what you weren't sure about.